### PR TITLE
Add Features implementation for plans

### PR DESCRIPTION
**Pull request by ZeeshanAzhar23**

### CODE REVIEW

File: app/controllers/features_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class FeaturesController < ApplicationController
+  before_action :find_feature, only: [:show,:edit,:destroy,:update]
```
Comment (javier459) [inline app/controllers/features_controller.rb]: spacing issue

File: app/controllers/features_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  def index
+  @features=Plan.find(params[:plan_id]).features
```
Comment (javier459) [inline app/controllers/features_controller.rb]: spacing issue

File: app/controllers/features_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+  def new
+    @plan = Plan.find(params[:plan_id])
+    @feature=@plan.features.new
```
Comment (javier459) [inline app/controllers/features_controller.rb]: spacing issue

File: app/controllers/features_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    authorize @feature
+    if @feature.update(params_features)
+      redirect_to @feature
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    id=@feature.plan_id
```
Comment (javier459) [inline app/controllers/features_controller.rb]: spacing issue

File: app/controllers/features_controller.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  end
+  private def params_features
+    params.require(:feature).permit(:name,:code,:unit_price,:max_unit_limit)
+  end
+
+  private def find_feature
+    @plan=Plan.find(params[:plan_id])
+    @feature=@plan.features.find(params[:id])
```
Comment (javier459) [inline app/controllers/features_controller.rb]: spacing issue

File: app/controllers/features_controller.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    else
+    render 'show'
+  end
+
+  end
+  private def params_features
+    params.require(:feature).permit(:name,:code,:unit_price,:max_unit_limit)
+  end
+
+  private def find_feature
```
Comment (javier459) [inline app/controllers/features_controller.rb]: convert into private block

File: app/models/feature.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  validates :name, presence: true, uniqueness: { case_sensitive: false }, length: {maximum: 20}
+  validates :code, presence: true, uniqueness: { case_sensitive: false }, length: {maximum: 20}
+  validates :unit_price, presence: true, length: {maximum: 10}
+  validates :max_unit_limit, presence: true,length: {maximum: 20}
+
+
+
+
+
+
```
Comment (javier459) [inline app/models/feature.rb]: remove extra empty lines

File: app/views/features/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      <td><%= feature.unit_price %></td>
+      <td><%= feature.max_unit_limit %></td>
+      <td><%= link_to 'Show', [feature.plan, feature] %></td>
+        <%if current_user && current_user.role == 'admin'%>
+        <td><%= link_to 'Edit', [:edit,feature.plan, feature] %></td>
+        <td><%= link_to 'Destroy', [feature.plan, feature], method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <%end %>
+      </tr>
+      <%count = count.next%>
+    <% end %>
```
Comment (javier459) [inline app/views/features/index.html.erb]: use partial here

**Review by javier459 — CHANGES_REQUESTED** [review]

**Review by javier459 — APPROVED** [review]

**Review by ZeeshanAzhar23 — COMMENTED** [review]
Kindly review the changes

**Pull request merged**